tools/sense_studio/train_module.py

```diff
@@ -0,0 +1,84 @@
+import os
+import subprocess
+import shlex
+import urllib
+import time
+
+import flask
+from flask import Blueprint
+from flask import render_template
+from flask import request
+from flask import current_app
+from flask import stream_with_context
+
+from tools.sense_studio import utils
+
+train_bp = Blueprint('train_bp', __name__)
+
+PROCESS = None
+
+
+@train_bp.route('/<string:project>', methods=['GET'])
+def training_page(project):
+    project = urllib.parse.unquote(project)
+    return render_template('train.html', project=project, models=utils.BACKBONE_MODELS, is_disabled=False)
+
+
+def stream_template(template_name, **context):
+    # Ref: https://flask.palletsprojects.com/en/1.1.x/patterns/streaming/
+    current_app.update_template_context(context)
+    t = current_app.jinja_env.get_template(template_name)
+    rv = t.stream(context)
+    return rv
+
+
+@train_bp.route('/train-model', methods=['POST'])
+def train_model():
+    data = request.form
+    project = data['project']
+    num_layers_to_finetune = data['layers_to_finetune']
+    path_out = data['output_folder']
+    model_name = data['model_name']
+    path = utils.lookup_project_path(project)
+    config = utils.load_project_config(path)
+
+    train_classifier = ["python tools/train_classifier.py", f"--path_in=/home/twentybn/Code/sense/dataset/{project}",
+                        f"--num_layers_to_finetune={num_layers_to_finetune}",
+                        "--use_gpu" if config['use_gpu'] else "",
+                        f"--path_out={path_out}" if path_out else "",
+                        f"--model_name={model_name}" if model_name else "",
+                        "--overwrite"]
+
+    train_classifier = ' '.join(train_classifier)
+    train_classifier = shlex.split(train_classifier)
+
+    global PROCESS
+    PROCESS = subprocess.Popen(train_classifier, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False)
```

Comment:
What happens if the user leaves the page, before training has ended and without clicking `Cancel`? Will there just be an invisible long-running process in the backend?

Can we maybe indicate on the UI that there is currently a training in progress so the user can go there to cancel it?

Reply:
I'll investigate on that but afaik if the user leaves that page the training process automatically gets cancelled. I have felt that but still will investigate.